src/components/icon/icon.pw.tsx

```diff
@@ -0,0 +1,548 @@
+        `${radius}%`
+      );
+      await expect(iconLocator).toHaveCSS(
+        "border-top-left-radius",
+        `${radius}%`
+      );
+      await expect(iconLocator).toHaveCSS(
+        "border-top-right-radius",
+        `${radius}%`
+      );
+    });
+  });
+
+  ([
+    [SIZE.EXTRASMALL, 16],
+    [SIZE.SMALL, 24],
+    [SIZE.MEDIUM, 32],
+    [SIZE.LARGE, 40],
+    [SIZE.EXTRALARGE, 56],
+  ] as [IconProps["bgSize"], number][]).forEach(([size, heightAndWidth]) => {
+    test(`should check bgSize as ${size} when height and width is set as ${heightAndWidth}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconComponent bgSize={size} />);
+
+      const iconLocator = page.getByTestId("icon");
+      await expect(iconLocator).toHaveCSS("height", `${heightAndWidth}px`);
+      await expect(iconLocator).toHaveCSS("width", `${heightAndWidth}px`);
+    });
+  });
+
+  ([
+    [["left"], "left", "bottom", 0, 0],
+    [["top"], "top", "bottom", 0, 0],
+    [["left"], "left", "top", 0, 120],
+    [["bottom"], "bottom", "top", 0, 120],
+    [["bottom"], "bottom", "left", 0, 120],
+    [["bottom"], "bottom", "right", 0, 120],
+    [["top"], "top", "left", 0, 0],
+    [["top"], "top", "right", 0, 0],
+    [["right"], "right", "bottom", 700, 0],
+    [["right"], "right", "top", 700, 120],
+  ] as [
+    IconProps["tooltipFlipOverrides"],
+    string,
+    IconProps["tooltipPosition"],
+    number,
+    number
+  ][]).forEach(
+    ([
+      flipPosition,
+      expectedPosition,
+      tooltipPosition,
+      horizontalPos,
+      verticalPos,
+    ]) => {
+      test(`should check tooltip position is ${expectedPosition} rather than ${tooltipPosition} when flip position is ${flipPosition} after scrolling`, async ({
+        mount,
+        page,
+      }) => {
+        await mount(
+          <div style={{ padding: "60px 60px 60px 60px" }}>
+            <IconTooltipComponent
+              tooltipFlipOverrides={flipPosition}
+              tooltipPosition={tooltipPosition}
+            />
+          </div>
+        );
+
+        await page.setViewportSize({ width: 700, height: 120 });
+
+        await page.evaluate(
+          ([horizontal, vertical]) => window.scrollTo(horizontal, vertical),
+          [horizontalPos, verticalPos]
+        );
+
+        const tooltip = getDataElementByValue(page, "tooltip");
+        await expect(tooltip).toHaveAttribute(
+          "data-placement",
+          expectedPosition
+        );
+      });
+    }
+  );
+});
+
+test.describe("should check accessibility for Icon component", () => {
+  [true, false].forEach((boolVal) => {
+    test(`should pass accessibility tests when aria-hidden prop is set as ${boolVal}`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<IconComponent aria-hidden={boolVal} />);
+
+      await checkAccessibility(page);
+    });
+  });
+
+  // test("should pass accessibility tests when aria-hidden prop is set as true", async ({
```

Comment:
**nitpick:** You have commented out tests here. Have they been left in by accident or commented out by accident?